lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

```diff
@@ -139,4 +206,48 @@ public JavaLaunchCommandBuilder enableAssertions(final boolean assertionsEnabled
     this.assertionsEnabled = assertionsEnabled;
     return this;
   }
+
+  /**
+   * Represents the JVM option as a option and value, combined by a separator.
+   * There are many different JVM option formats. This implementation only recognizes
+   * equals-separated and -Xm[nsx] memory options. All other option formats are
+   * represented with an option and empty value and separator.
+   */
+  static final class JVMOption {
```

Comment:
Even though we're only comparing `this.option` for equality right now, probably a good idea to implement `equals()` and `hashCode()` since this is a value class and we could easily be comparing entire objects for equality in the future (e.g., if they were part of a `Set`).